earth2mip/lagged_ensembles/score.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+import torch
+
+import earth2mip.grid
+from earth2mip.crps import crps_from_empirical_cdf
+
+logger = logging.getLogger(__name__)
+
+
+def weighted_average(x, w, dim):
+    return torch.mean(x * w, dim) / torch.mean(w, dim)
+
+
+def area_average(grid: earth2mip.grid.LatLonGrid, x: torch.Tensor):
+    lat = torch.tensor(grid.lat, device=x.device)[:, None]
+    cos_lat = torch.cos(torch.deg2rad(lat))
+    return weighted_average(x, cos_lat, dim=[-2, -1])
+
+
+def score(
```

Comment:
Does this function need to assume the use of lagged ensembles? Is there anyway to make a more general `score` function and one way to use it would be to plug in lagged ensembles? Again, I think we should deisolate much of the code here to be more broadly usable.

Reply:
It's specific to lagged ensembles in that the m=0 lag is considered the deterministic skill.

But, this is a change of the implementation of scoring for the lagged ensemble script, and not public API. Perhaps it could be generalized elsewhere, but I think it is out of scope of the PR or the original usage of this code.

Reply:
I agree that it can be out of scope for this PR but we should create an issue to make the scoring function more general.

I can be persuaded that we do actually need multiple score functions like so:
```
earth2mip.lagged_ensemble.score
earth2mip.s2s.score
...
```

Reply:
resolving, since out of scope.